test/Conversion/TorchToMhlo/reduction.mlir

```diff
@@ -1,239 +0,0 @@
-// RUN: torch-mlir-opt <%s -convert-torch-to-mhlo -split-input-file -verify-diagnostics | FileCheck %s
```

Comment:
Are these tests not needed anymore?

Reply:
The architecture notes suggest not using unit tests "for these "macro expansion" patterns", https://github.com/llvm/torch-mlir/blob/main/docs/architecture.md#dos-and-donts-for-unit-vs-end-to-end-testing